includes/class-sensei-learner.php

```diff
@@ -111,6 +114,40 @@ public function delete_all_user_activity( $user_id = 0 ) {
 		return $dataset_changes;
 	}
 
+	/**
+	 * Remove duplicate progress comments to mitigate duplicate enrollment issue.
+	 *
+	 * Hooked into sensei_log_activity_after
+	 *
+	 * @since 3.7.0
+	 * @access private
+	 *
+	 * @param array $args
+	 */
+	public function remove_duplicate_progress( $args ) {
+		add_action(
+			'shutdown',
+			function() use ( $args ) {
+				global $wpdb;
+
+				// This query remove all comments, but the first one, which match the context conditions.
+				// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+				$wpdb->query(
+					$wpdb->prepare(
+						"DELETE c1.*
+							FROM $wpdb->comments as c1
+							INNER JOIN
+								( SELECT comment_ID FROM $wpdb->comments WHERE comment_post_ID = %d AND user_id = %d AND comment_type = %s ORDER BY comment_ID LIMIT 1, 99 ) c2
```

Comment:
`LIMIT 1, 99` I used it to make the query simpler, and it's unlikely we'll have 99 duplicate progress.